### ISSUE TITLE
Regression on `Type.Partial` in conjunction with `Type.Record(Type.Enum(xyz), T)` since 0.31.20

### ISSUE THREAD

**Issue by mpsq** [sinclairzx81/typebox#651]
Since https://github.com/sinclairzx81/typebox/pull/647 (0.31.20), the following:

```typescript
enum Keys {
  Name = "name",
  Description = "description",
}
const Schema = Type.Partial(Type.Record(Type.Enum(Keys), Type.Unknown()))
const object: Static<typeof Schema> = {
  [Keys.Name]: "123",
}
```

Errors with: 

```
 [typescript] Property '[Keys.Description]' is missing in type '{ name: string; }' but required in type 'Record<Keys, unknown>'.
```

cc. @quentinverlhac

**Comment by sinclairzx81** [comment]
@mpsq Hi,

Have published a fix for this on 0.31.22. This update changes the return type of Record to be TObject when specifying keys of type TEnum (allowing the returned Object to be used with Partial). This works similar to passing a Union of Literals for the key where a fixed sized TObject is returned. Note that the enum inference path is relatively new (was omitted and only re-introduced on 0.31.20), so having this function return TObject with known properties for TEnum keys makes sense here I think.

[TypeScript Link Here](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgZRgQxsAxgGjgFQE8wBTOAXzgDMoIQ4ByAAQGdgA7LAGzWCgHoYxEgCMIADwYBuAFAyS7AK70A0iUItEMuHAByaEGQC8cAETsDJUzm1wAIiRZYowMJgjs4J0wBNHz13d2axlyGSwPFnhkLAALEhA0LwJhADoABTRYYDQuAAoiUlSAJRIIqB8CtIBRJRA8tQ0ASjxCklSAVXYAa3YIAHd2PKaR8Mj4CBEAKzKYAC4UdEwsAB4hUggqFDiEtAA+ZIRbAG1GllT9QwBdBdMARgAmAGYQ8iA) (note: your repro should work if you clear the cache on the TSP and ensure the latest TB is pulled down)

The PR automatically closed the issue, but if you have any problems, feel free to let me know and will re-open.
Thanks for reporting!
All the best
S

**Comment by mpsq** [comment]
Thanks for such a quick fix & reply :bow:
I just pulled 0.31.22, I can confirm that the regression is fixed :pray:
